model-engine/tests/unit/domain/test_llm_use_cases.py

```diff
@@ -869,3 +870,60 @@ async def test_download_nonexistent_model_raises_not_found(
     )
     with pytest.raises(ObjectNotFoundException):
         await use_case.execute(user=user, request=request)
+
+
+@pytest.mark.asyncio
+async def test_delete_model_success(
+    fake_model_endpoint_service,
+    fake_llm_model_endpoint_service,
+    llm_model_endpoint_sync: Tuple[ModelEndpoint, Any],
+    test_api_key: str,
+):
+    fake_llm_model_endpoint_service.add_model_endpoint(llm_model_endpoint_sync[0])
+    fake_model_endpoint_service.add_model_endpoint(llm_model_endpoint_sync[0])
+    use_case = DeleteLLMEndpointByNameUseCase(
+        model_endpoint_service=fake_model_endpoint_service,
+        llm_model_endpoint_service=fake_llm_model_endpoint_service,
+    )
+    user = User(user_id=test_api_key, team_id=test_api_key, is_privileged_user=True)
+    response = await use_case.execute(
+        user=user, model_endpoint_name=llm_model_endpoint_sync[0].record.name
+    )
+    assert response.deleted is True
+
+
+@pytest.mark.asyncio
+async def test_delete_nonexistent_model_raises_not_found(
+    fake_model_endpoint_service,
+    fake_llm_model_endpoint_service,
+    llm_model_endpoint_sync: Tuple[ModelEndpoint, Any],
+    test_api_key: str,
+):
+    fake_llm_model_endpoint_service.add_model_endpoint(llm_model_endpoint_sync[0])
+    fake_model_endpoint_service.add_model_endpoint(llm_model_endpoint_sync[0])
+    use_case = DeleteLLMEndpointByNameUseCase(
+        model_endpoint_service=fake_model_endpoint_service,
+        llm_model_endpoint_service=fake_llm_model_endpoint_service,
+    )
+    user = User(user_id=test_api_key, team_id=test_api_key, is_privileged_user=True)
+    with pytest.raises(ObjectNotFoundException):
+        await use_case.execute(user=user, model_endpoint_name="nonexistent-model")
+
+
+@pytest.mark.asyncio
+async def test_delete_unauthorized_model_raises_not_authorized(
```

Comment:
should we also have a test that a public model can't be deleted by anyone but the owner?

Reply:
yes, will add now

Reply:
Actually, I think this test is actually subsumed by `test_delete_unauthorized_model` since no one but the owner would be authorized to delete any model - do you think there should be another separate test?

Reply:
yeah, I was thinking that we'd probably add a test to potentially catch a case in the future where we change some authorization logic that ends up using the `public_model` field, and we inadvertently allow anyone to delete public models

Reply:
Where is the `public_model` field? Do you mean I should add a field called `public_model` to `ModelEndpointRecord`?

Reply:
per discussion: ensure that there is a test that checks the `public_inference` field is True and when a user who has team access but not ownership of a model tries to delete it, the deletion fails.

Reply:
I looked into this test more, and we currently determine write access based on the condition `return user.team_id == owned_entity.owner`, so either I should add a new authz method that checks that the `user.user_id == owned_entity.owner` or leave this as is and make the test with a different user_id and team_id than the owner of the endpoint. Currently, anyone who is under the same team_id as the initial model owner has the ability to delete the model. Is this not the desired end state?

Reply:
It's fine to keep the check as `user.team_id == owned_entity.owner`, we've been assuming that people on the same team have full access to each other's resources (it's all owned by the team anyways), so we can keep it this way